src/Plugin/Hal.php

```diff
@@ -605,6 +605,14 @@ public function renderCollection(Collection $halCollection)
             $payload['page'] = ($payload['page_count'] > 0)
                 ? $halCollection->getPage()
                 : 0;
+            $payload['start_index'] = $payload['page'] > 0
+                ? ($payload['page_size'] * ($payload['page'] - 1)?:1)
+                : 0;
+            $payload['end_index'] = $payload['page'] > 0
+                ? ($payload['page'] < $payload['page_count']
+                    ? (($payload['page_size'] * $payload['page']) - 1)
+                    : $payload['total_items'])
+                : 0;
```

Comment:
This is pretty convoluted and hard to follow. Maybe write a dedicated method for calculating the value? That way you can return early when specific conditions are met.

Reply:
Do you think it's should be better to get values from Paginator instead of Hal plugin?

Reply:
IIRC, we use the values in the zf-hal configuration to seed the paginator. That said, the total number of items and number of pages is generally from the paginator, so getting those from that would likely be the best idea.